scala/sort/merge_sort.sc

```diff
@@ -0,0 +1,31 @@
+object MergeSort {
+  def apply(comparisonPredicate: (Int, Int) => Boolean)(seq: Seq[Int]): Seq[Int] = {
+    seq match {
+      case first +: second +: tail =>
+        val pivot = Math.round((seq.length - 1).toFloat / 2)
+        merge(
+          comparisonPredicate,
+          apply(comparisonPredicate)(seq.take(pivot)),
+          apply(comparisonPredicate)(seq.drop(pivot))
+        )
+      case _ => seq
+    }
+  }
+
+  private def merge(predicate: (Int, Int) => Boolean, left: Seq[Int], right: Seq[Int]): Seq[Int] = {
+    left.length match {
+      case 0 => right
+      case _ => right.length match {
+        case 0 => left
+        case _ =>
+          val headLeft +: tailLeft = left;
```

Comment:
Perhaps we can call `tailLeft` as `remainderLeft` or `remLeft` so that it's clear its a collection of elements rather than referring only to the tail element?

Reply:
Just realized @varunganesh91 used the same `tail` to refer to a list of remaining elements (https://github.com/nugit/LYANL/pull/2/files) 😂 

Is this a common concept? 🤔

Reply:
It is :) In fact:
```
scala> List(1,2,3).tail
res1: List[Int] = List(2, 3)
```

Reply:
🤔,👌